dev-support/jenkins-scripts/generate-hbase-website.sh

```diff
@@ -23,73 +23,173 @@
 #
 # It needs to be built on a Jenkins server with the label git-websites
 #
-# It expects to have the hbase repo cloned to the directory hbase
+# Allows specifying options for working directory, maven repo, and publishing to git
+# run with --help for usage.
 #
 # If there is a build error, the Jenkins job is configured to send an email
 
-LOCAL_REPO=${WORKSPACE}/.m2/repo
-# Nuke the local maven repo each time, to start with a known environment
-rm -Rf "${LOCAL_REPO}"
-mkdir -p "${LOCAL_REPO}"
+declare CURRENT_HBASE_COMMIT
+declare PUSHED
+declare FILE
+declare WEBSITE_COMMIT_MSG
+declare -a FILES_TO_REMOVE
 
-# Clean any leftover files in case we are reusing the workspace
-rm -Rf -- *.patch *.patch.zip hbase/target target *.txt hbase-site
+set -e
+function usage {
+  echo "Usage: ${0} [options] /path/to/hbase/checkout"
+  echo ""
+  echo "    --working-dir /path/to/use  Path for writing logs and a local checkout of hbase-site repo."
+  echo "                                if given must exist."
+  echo "                                defaults to making a directory in /tmp."
+  echo "    --local-repo /path/for/maven/.m2  Path for putting local maven repo."
+  echo "                                if given must exist."
+  echo "                                defaults to making a clean directory in --working-dir."
+  echo "    --publish                   if given, will attempt to push results back to the hbase-site repo."
+  echo "    --help                      show this usage message."
+  exit 1
+}
+# if no args specified, show usage
+if [ $# -lt 1 ]; then
+  usage
+fi
+
+# Get arguments
+declare component_dir
+declare working_dir
+declare local_repo
+declare publish
+while [ $# -gt 0 ]
+do
+  case "$1" in
+    --working-dir) shift; working_dir=$1; shift;;
+    --local-repo) shift; local_repo=$1; shift;;
+    --publish) shift; publish="true";;
+    --) shift; break;;
+    -*) usage ;;
+    *)  break;;  # terminate while loop
+  esac
+done
+
+# should still have where component checkout is.
+if [ $# -lt 1 ]; then
+  usage
+fi
+component_dir="$(cd "$(dirname "$1")"; pwd)/$(basename "$1")"
+
+if [ -z "${working_dir}" ]; then
+  working_dir=/tmp
+  while [[ -e ${working_dir} ]]; do
+    working_dir=/tmp/hbase-generate-website-${RANDOM}.${RANDOM}
```

Comment:
Maybe emit that we are using a default value, for every time we use a default value because we didn’t find a flag?